src/components/Icon/BankIcons.ts

```diff
@@ -1,158 +1,107 @@
+        [CONST.BANK_NAMES.HUNTINGTON_BANK]: isCard
+            ? (require('@assets/images/cardicons/huntington-bank.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/huntington-bank.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.NAVY_FEDERAL_CREDIT_UNION]: isCard
+            ? (require('@assets/images/cardicons/navy-federal-credit-union.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/navy-federal-credit-union.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.PNC]: isCard ? (require('@assets/images/cardicons/pnc.svg') as React.FC<SvgProps>) : (require('@assets/images/bankicons/pnc.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.REGIONS_BANK]: isCard
+            ? (require('@assets/images/cardicons/regions-bank.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/regions-bank.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.SUNTRUST]: isCard
+            ? (require('@assets/images/cardicons/suntrust.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/suntrust.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.TD_BANK]: isCard
+            ? (require('@assets/images/cardicons/td-bank.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/td-bank.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.US_BANK]: isCard
+            ? (require('@assets/images/cardicons/us-bank.svg') as React.FC<SvgProps>)
+            : (require('@assets/images/bankicons/us-bank.svg') as React.FC<SvgProps>),
+        [CONST.BANK_NAMES.USAA]: isCard ? (require('@assets/images/cardicons/usaa.svg') as React.FC<SvgProps>) : (require('@assets/images/bankicons/usaa.svg') as React.FC<SvgProps>),
+    } as const;
+
+    // Fallback to generic bank/card icon
+    const iconModule =
+        iconMappings[bankValue] ||
+        (isCard ? (require('@assets/images/cardicons/generic-bank-card.svg') as React.FC<SvgProps>) : (require('@assets/images/bankicons/generic-bank-account.svg') as React.FC<SvgProps>));
+    return iconModule;
+}
 
-    return isCard ? GenericBankCard : GenericBank;
+function getBankNameKey(bankName: string): BankNameKey {
+    const bank = Object.entries(CONST.BANK_NAMES).find(([, value]) => value?.toLowerCase() === bankName);
+    return (bank?.[0] as BankNameKey) ?? '';
 }
 
 /**
  * Returns Bank Icon Object that matches to existing bank icons or default icons
  */
-
 export default function getBankIcon({themeStyles, bankName, isCard = false}: BankIconParams): BankIcon {
     const bankIcon: BankIcon = {
         icon: isCard ? GenericBankCard : GenericBank,
     };
-
     if (bankName) {
-        bankIcon.icon = getAssetIcon(bankName.toLowerCase(), isCard);
+        const bankNameKey = getBankNameKey(bankName.toLowerCase());
+
+        if (bankNameKey && Object.keys(CONST.BANK_NAMES).includes(bankNameKey)) {
+            bankIcon.icon = getAssetIcon(bankNameKey, isCard);
```

Comment:
App crashes here

<img width="1483" alt="crash" src="https://github.com/Expensify/App/assets/108292595/b26a4cf0-7484-4bb5-acb5-bf2117f5f1d2">


For easy testing, hardcode like this: `bankIcon.icon = getAssetIcon('CHASE', false)` before `return bankIcon;`